source/sessions/driver-sessions.rst

```diff
@@ -502,12 +504,12 @@ include a session ID in a ``KILLCURSORS`` command.
 
 Sessions and Connections
 ========================
-A driver MUST only obtain an implicit session after it successfully checks out a connection.
-This limits the number of implicit sessions to never exceed the maximum connection pool size.
-The motivation for this behavior is to prevent too many sessions from being created in a scenario
-where only a limited number are actually needed to execute operations (i.e. TooManyLogicalSessions error).
+A driver MUST only obtain an implicit session's ``ServerSession`` after it successfully checks out a connection.
+A driver SHOULD NOT attempt to release the acquired session before connection check in.
+Through both through the pooling mechanism and limiting acquisition to a successful
```

Comment:
This sentence doesn't make sense to me. What exactly are we trying to say? Also, the motivation in the original text was clearly stated, and, I think helpful. So let's try to keep it, maybe clarifying that we are trying to limit the number of *server* sessions created.

Reply:
Agreed. Maybe we can remove this sentence and add a bit of motivation to the first sentence:
> To reduce the number of ``ServerSession`` created, the driver MUST only obtain an implicit session's ``ServerSession`` after it successfully checks out a connection.

Reply:
Yea that is more concise, moved!